docs/extensions/controllerregistration.md

```diff
@@ -154,3 +169,38 @@ In the above example, Gardener itself does not understand the AWS-specific provi
 However, if this part of the `Shoot` resource should be validated then you should run an AWS-specific component in the garden cluster that registers a webhook. You can do it similarly if you want to default some fields of a resource (by using a `MutatingWebhookConfiguration`).
 
 Again, similar to how Gardener is deployed to the garden cluster, these components must be deployed and managed by the Gardener administrator.
+
+### `Extension` resource configurations
+
+The `Extension` resource allows to inject arbitrary steps into the shoot reconciliation flow that are unknown to Gardener.
+Hence, it is slightly special and allows further configuration when registering it:
+
+```yaml
+apiVersion: core.gardener.cloud/v1beta1
+kind: ControllerRegistration
+metadata:
+  name: extension-foo
+spec:
+  resources:
+  - kind: Extension
+    type: foo
+    primary: true
+    globallyEnabled: true
+    reconcileTimeout: 30s
+```
+
+The `globallyEnabled=true` option specifies that the `Extension/foo` object shall be created by default for all shoots (unless they opted out by setting `.spec.extensions[].enabled=false` in the `Shoot` spec).
+
+The `reconcileTimeout` tells Gardener how long it should wait during its shoot reconciliation flow for the `Extension/foo`'s reconciliation to finish.
```

Comment:
I just double checked and realized that this setting does not have any affect at the moment. Since you're already improving the extension handling, can you add the timeout parameter to the extension deployment [here](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/extension.go#L79) and to the deletion [here](https://github.com/gardener/gardener/blob/master/pkg/operation/botanist/extension.go#L126). Otherwise I will open a separate PR if you prefer to not do it here.

Reply:
I'll pass it in the wait-for-reconcile-ready function, but I don't know how to pass it in the wait-for-deletion case (second link you shared). We don't read the timeout yet, we simply wait all existing `Extension` resources with the same timeout. It's not straight-forward to fix this IMO, because when the respective `ControllerRegistration` is changed (supported for the `Extension` kind/type is removed, for example), then also the information about the `reconcileTimeout` is gone. However, as the field is already called `*reconcile*Timeout` it's maybe fair for now to only use it during the reconciliation? If you still want to fix this (maybe the last known configured timeout should be stored as annotation on the `Extension` object directly (to not lose the information), or so ..) then I suggest to follow up with it separately.

Reply:
Makes sense, thank you 👍